ci-scripts/JenkinsFile-build-vagrant-boxes

```diff
@@ -0,0 +1,104 @@
+#!/bin/groovy
+/*
+ * Licensed to the OpenAirInterface (OAI) Software Alliance under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The OpenAirInterface Software Alliance licenses this file to You under
+ * the terms found in the LICENSE file in the root of this
+ * source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *-------------------------------------------------------------------------------
+ * For more information about the OpenAirInterface (OAI) Software Alliance:
+ *      contact@openairinterface.org
+ */
+
+
+def GIT_URL = "github.com/magma/magma"
+def GIT_COMMIT
+
+pipeline {
+  agent {
+    label "buildPacketSlave"
+  }
+  parameters {
+    string(name: 'branch', defaultValue: 'master')
+  } 
+  options {
+    disableConcurrentBuilds()
+    timestamps()
+    ansiColor('xterm')
+  }
+  stages {
+    stage ("Retrieve and Prepare Source Code") {
+      steps {
+        script {
+          cleanWs()
+          checkout( 
+            changelog: false, 
+            poll: false, 
+            scm: [
+            $class: 'GitSCM', 
+            branches: [[name: params.branch]], 
+            doGenerateSubmoduleConfigurations: false, 
+            extensions: [], 
+            submoduleCfg: [], 
+            userRemoteConfigs: [[refspec: '+refs/pull/*:refs/remotes/origin/pr/*', url: "https://" + GIT_URL + ".git"]]]
+          )
+          GIT_COMMIT = sh returnStdout: true, script: 'git log -1 --pretty="%H"'
+        }
+      }
+    }
+    stage ("Build magma-dev") {
+      environment {
+        // TODO(119vik): Replace with Jenkins.Secret
+        ATLAS_TOKEN = "123"
```

Comment:
Are we using Jenkins inbuilt secret manager? Just wondering

Reply:
We should store secrets in Jenkins credentials store. Otherwise, we would expose it to users who read logs.